### DISCUSSION TITLE
Apparent Temperature- Source?

### DISCUSSION THREAD

**Discussion by gary-hammer** [Daveiano/weewx-wdc#158] [Q&A] [answered]
Just wondering how apparent temperature is derived.

**Comment by Daveiano** [comment]
Here is the formula: http://www.bom.gov.au/info/thermal_stress/#atapproximation

It uses temperature, relative humidity and wind speed for the calculation.

Here is the formula implementation in weewx: https://github.com/weewx/weewx/blob/master/bin/weewx/wxformulas.py#L523

Reference: https://groups.google.com/g/weewx-user/c/bNz3ymPNHHc/m/VuHcJAc1AwAJ

**Comment by gary-hammer** [comment]
Thanks.
I suspect it is the same as the weatherlink value for THSW

I like this explainer from Davis.

https://www.davisinstruments.com/blogs/newsletter/meteorology-101-understanding-indexes

**Comment by Daveiano** [comment]
Yes, it sounds like the same (the formula which uses radiation).

Here is another discussion started by a Vantage Vue user: https://groups.google.com/g/weewx-user/c/ARBHWDXiDpQ/m/pyHzFA89BAAJ

**Comment by gary-hammer** [comment]
So, now that I know it is the THSW value, how can I change that to use another value? At my location, the simple heat index is much closer to what it feels like outside the house. I'd perhaps change it to wind chill in the late fall as well as THSW is not useful at all during winter.

**Comment by Daveiano** [comment]
I think you can not change how the apparent temperature is calculated (you would need to change the WeeWX Code I guess). What you can do is only showing the values you are interested in: 

You could hide `appTemp` and `windchill` and only show `heatindex`? 

Just remove `appTemp` and `windchill` from `DisplayOptions > stat_tile_observations`. Also remove them from every context in `DisplayOptions > diagrams > day, week, month, etc`

You could also try and change the labels of the observations to kind of switch them, this can be achieved by editing the translation files (lang/en.conf)

**Comment by gary-hammer** [comment, marked as answer]
Well! So I was confused as to why my apptemp equaled the thsw from the WLL.
I went to https://github.com/weewx/weewx/blob/6c90a174f9cf81b1b93b83b70a426199a51b47ba/bin/weewx/wxformulas.py
The apptemp is a calculated value and from the code, should not equal the thsw value from the WLL.
I looked through my weewx.conf and found that I had mapped the incoming WLL value thsw to apptemp.
Doesn't matter if it should be calculated, that mapping takes precedence. The calculated apptemp is a much more realistic value for my location/use that any of the other calculated temps.

So, in short, user error. But thanks for prompting me to dig deeper. I like to understand what's going on under the hood, not just install something and never learn about it.
